# Paper Introduction Rewrite #

For my first paper that targets a broad rather than a specialized audience, I chose the [Common SNPs explain a large proportion of the heritability for human height paper][1] that appeared in Nature. I rewrote the beginning of the introduction to immediately discuss the problem and proposed solution.

_SNPs discovered by genome-wide association studies (GWASs), sets of univariate association tests between loci and phenotype scaled by the number of tests performed, account for only a small fraction of the genetic variation of complex traits in human populations. Even when considering narrow-sense heritability, so that non-additive effects are disregarded, the validated SNPs fail to account for the vast portion of phenotypical variance. This can be explained by two reasons: (1) each of causal variants explain such a small amount of variance that they do not reach stringent FDR or Bonferroni significant levels and/or (2) the causal variants are not in complete linkage disequilibrium with genotyped SNPs. Our approach attempts to solve these problems by jointly considering the variance explained by SNPs, instead of individually. We then test our approach on human height, a highly heritable trait (estimated to be ~0.8), showing that increases the variance explained by the identifed SNPs significantly._


For my second paper that targets a specialized audience, I chose the [Inference of Population Structure Using Multilocus Genotype Data paper][2] that appeared in Genetics. I rewrote the beginning of the introduction of the paper to broaden its focus towards the problem and lessen its focus on the biology and methods. 

_Understanding population structures is important in identifying evolutionary processes over time and how people have migrated and mixed. Also, modeling population structures allows individuals of unknown origin to uncover their ancestry. Naturally, scientists have attempted to uncover these population structures by sampling DNA from a set of individuals and then estimating parameters about the underlying populations. Then, using these parameters, specifically allele frequencies at different loci, scientists can estimate the likelihood of new individuals of unknown origins to any of the populations. However, there are a number of challenges associated with building such a model and learning its parameters. First, defining the populations that the sequenced individuals come from is difficult. Population definition is generally subjective based on phsyical or cultural characteristics, but there is little proof of consistencies of these observations with genetic information. Second, individuals can come from more than one population (i.e. mixed), so a hard assignment of population when learning is impractical and inaccurate. We propose a Bayesian clustering approach to this problem that assumes little about the populations of the sampled individuals and learns the population parameters and the assignments of individuals probabilistically._

[1]: https://d1b10bmlvqabco.cloudfront.net/attach/j6zot2yz1ti44r/j77pp0la26p6x9/j7gmb55ttnzv/ng.608.pdf
[2]: http://www.genetics.org/content/155/2/945
